pkg/ingester/ingester.go

```diff
@@ -632,11 +627,8 @@ func (i *Ingester) starting(ctx context.Context) (err error) {
 		servs = append(servs, services.NewIdleService(nil, nil))
 	}
 
-	i.subservices, err = services.NewManager(servs...)
-	if err == nil {
-		err = services.StartManagerAndAwaitHealthy(ctx, i.subservices)
-	}
-	return errors.Wrap(err, "failed to start ingester components")
+	i.subservicesAfterIngesterRingLifecycler, err = createManagerThenStartAndAwaitHealthy(ctx, servs...)
```

Comment:
What if instead of building `servs` slice, we passed all individual and possibly `nil` services to `createManagerThenStartAndAwaitHealthy`, which would then filter out `nil` values and add idle service if neccessary? This would simplify `starting` method a bit, by moving more logic to `createManagerThenStartAndAwaitHealthy`.

Reply:
I don't think we can completely avoid `if` conditions, but we can simplify it: https://github.com/grafana/mimir/pull/7769/commits/66dcf58a121d84684de1e25b738b2ed030151a90

Reply:
I reverted it, because of nil comparison with interface. Solving it is possible but I think makes the code even more complicated, so I will move on on this.